iocage_lib/ioc_common.py

```diff
@@ -1078,3 +1078,42 @@ def get_host_gateways():
             gateways[af_mapping[af]]['interface'] = \
                 default_route[0]['interface-name']
     return gateways
+
+
+def parse_nat_fwds(forwards, log=None):
+    if log:
+        log.debug(f'Parsing NAT forwards: {forwards}')
+
+    for fwd in forwards.split(','):
+        proto, port = fwd.split('(')
+        port = port.strip('()')
+
+        if log:
+            log.debug(f'Proto: {proto} Port: {port}')
+        try:
+            port, map = port.rsplit(':', 1)
+        except ValueError:
+            map = port
+
+        if log:
+            log.debug(f'Mapping {port} to {map}')
+
+        yield proto, port, map
+
+
+def get_jails_with_config(filters=None, mapping_func=None):
```

Comment:
Why do this instead of just using ioc_json? That's the good place to put this :P

Reply:
That would mean getting list of jails which exist and then retrieving their config, why duplicate code when we already are doing that in iocage_lib.iocage ?

Reply:
We're duplicating code here, the lib's usage is a small wrapper. It helps avoid a recursive dep

Reply:
I am sorry but that solution is way more bad then this one. We would need to use iocage.list to first retrieve list of datasets and once being done with that retrieve props not to mention corrupt jails. I would rather personally stick to a recursive dep :p Essentially the whole functionality of iocage.list + iocage.get ( all, non recursive ) - a good number of lines. Please advise @skarekrow @william-gr .

Reply:
Heh, that's fine then. It will be restructured eventually anyways.

Reply:
Yup ;)